public/app/features/dashboard-scene/scene/setDashboardPanelContext.ts

```diff
@@ -11,7 +11,14 @@ import { getDashboardSceneFor, getPanelIdForVizPanel, getQueryRunnerFor } from '
 import { DashboardScene } from './DashboardScene';
 
 export function setDashboardPanelContext(vizPanel: VizPanel, context: PanelContext) {
-  context.app = CoreApp.Dashboard;
+  const dashboard = getDashboardSceneFor(vizPanel);
+  context.app = dashboard.state.isEditing ? CoreApp.PanelEditor : CoreApp.Dashboard;
+
+  const _onEnterEditMode = dashboard.onEnterEditMode;
+  dashboard.onEnterEditMode = () => {
+    _onEnterEditMode();
+    context.app = dashboard.state.isEditing ? CoreApp.PanelEditor : CoreApp.Dashboard;
+  };
```

Comment:
What about something like this @torkelo?
```suggestion
  const dashboard = getDashboardSceneFor(vizPanel);
  dashboard.subscribeToState(state => {
    if (state.isEditing) {
      context.app = CoreApp.PanelEditor;
    } else {
      context.app = CoreApp.Dashboard;
    }
  });
```

Reply:
thanks, yeah. that's cleaner.

still with Scenes it shows the suggestions (which don't work) once you go back to dashboard but have not technically exited the dashboard "edit" mode.

i think that's okay to leave for follow-up. i'd rather land this before 11.6 :)